### PR TITLE
Fds 47 tests

### PR DESCRIPTION
**Pull request by TimofeiShorokhov**
Task fds-47 with unit tests

### CODE REVIEW

**Comment by MaksimDanilau** [comment]
изображение

Meaningless commit messages. Please, don't follow these practices in the future.

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [inline Controllers/DeliveryController.go]: Just `GetOrder` as func name is fine. An order is already one, we don't need to explain this in the method name.

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [inline Controllers/DeliveryController.go]: It would be nice to implement `w.Header().Set("Content-Type", "application/json")` somewhere once.
Mostly every response in the application would be a JSON document.

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [inline Controllers/DeliveryController.go]: Unused global variable.

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [inline Controllers/DeliveryController.go]: `.Encode` Here is an unhandled error. It's not the Go way.

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [inline Controllers/DeliveryController.go]: What is `l` here? A variable name should have a meaning, not just `l`.

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [inline Controllers/DeliveryController.go]: Here you just forward the service (DB) error to the client of your API.

What the FE developer or QA should do when he received `8507: ORA-01422: exact fetch returns more than requested number of rows` for example?

You should wrap your lower-level error with some business error containing any piece of useful information.

Read this for more information https://go.dev/blog/go1.13-errors

Also, HTTP 400 is wrong here.

**Review by MaksimDanilau — CHANGES_REQUESTED** [review]
Iteration Number 1

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [reply Controllers/DeliveryController.go]: ✅

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [reply Controllers/DeliveryController.go]: ✅

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [reply Controllers/DeliveryController.go]: ✅

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [reply Controllers/DeliveryController.go]: ✅

File: Controllers/DeliveryController.go
Comment (MaksimDanilau) [reply Controllers/DeliveryController.go]: ✅

File: Controllers/DeliveryController.go
Comment (Ann40a) [inline Controllers/DeliveryController.go:30]: It is already server error, not bad request

File: Models/DeliveryModel.go
Comment (Ann40a) [inline Models/DeliveryModel.go:22]: When you return wrapped error you need to provide some extra context around it, otherwise it is unclear why to wrap it if you can just return initial error

File: db/DeliveryDB.go
Comment (Ann40a) [inline db/DeliveryDB.go:32]: better not to use `select *` when you need a specific order of fields, because some time later it could happen that multiple db migrations with alter table changed fields differently for different environments and now you couldn't rely on the order anymore

File: db/DeliveryDB.go
Comment (Ann40a) [inline db/DeliveryDB.go:28]: Why do you need to pass orders as input parameter instead of returning it from function?

File: other/other.go
Comment (Ann40a) [inline other/other.go:8]: Are these function still needed given that you switched to Gin? At least I couldn't find usages of it in this PR. Correct me if I'm wrong.

**Review by Ann40a — APPROVED** [review]
I've approved this PR. You can fix it and merge without waiting for one more review. Or you can even fix in a separate PR.

**Comment by TimofeiShorokhov** [comment]
all fixes in Tim_branch

**Pull request closed without merging**
